src/remarx/sentence/corpus/tei_input.py

```diff
@@ -154,12 +231,26 @@ def __post_init__(self) -> None:
 
     def get_text(self) -> Generator[dict[str, str]]:
         """
-        Get document content as plain text. Chunked by page, dictionary
-        includes page number.
+        Get document content as plain text. The document's content is yielded in segments
+        with each segment corresponding to a dictionary of containing its text content,
+        page number and section type ("text" or "footnote").
+        Body text is yielded once per page, while each footnote is yielded individually.
 
-        :returns: Generator with a dictionary of text content by page,
-        with page number.
+        :returns: Generator with dictionaries of text content, with page number and section_type ("text" or "footnote").
         """
-        # yield body text content chunked by page with page number
+        # yield body text and footnotes content chunked by page with page number
         for page in self.xml_doc.pages:
-            yield {"text": str(page), "page_number": page.number}
+            body_text = page.get_body_text()
+            if body_text:
+                yield {
+                    "text": body_text,
+                    "page_number": page.number,
+                    "section_type": SectionType.TEXT.value,
+                }
+
```

Comment:
Please add a comment here with a note explaining why we're yielding individual footnotes rather than one chunk of footnote content.  (I missed this convo & decision, but also seems important to remember for future revisions.)